Lib/importlib/_bootstrap_external.py

```diff
@@ -1635,3 +1635,29 @@ def _install(_bootstrap_module):
     supported_loaders = _get_supported_file_loaders()
     sys.path_hooks.extend([FileFinder.path_hook(*supported_loaders)])
     sys.meta_path.append(PathFinder)
+
+
+def alternative_architectures(suffixes):
+    """Add a suffix with an alternative architecture name
+    to the list of suffixes so an extension built with
+    the default (upstream) setting is loadable with our Pythons
+    """
+
+    arch_map = {
+        "arm-linux-gnueabi": "arm-linux-gnueabihf",
+        "armeb-linux-gnueabi": "armeb-linux-gnueabihf",
+        "mips64-linux-gnu": "mips64-linux-gnuabi64",
+        "mips64el-linux-gnu": "mips64el-linux-gnuabi64",
+        "ppc-linux-gnu": "powerpc-linux-gnu",
+        "ppc-linux-gnuspe": "powerpc-linux-gnuspe",
+        "ppc64-linux-gnu": "powerpc64-linux-gnu",
+        "ppc64le-linux-gnu": "powerpc64le-linux-gnu",
+    }
+
+    for suffix in suffixes:
+        for original, alternative in arch_map.items():
+            if original in suffix:
```

Comment:
How does this `suffix` usually look like? Is `in` safe? Can it do any harm if partial strings are matched here? E.g. when `'ppc-linux-gnu' in 'ppc-linux-gnuspe'`?

Reply:
Fixed. I forget to add dash as a prefix and dot as a suffix to the condition.